frontend/public/components/utils/firehose.jsx

```diff
@@ -129,8 +135,36 @@ export const Firehose = connect(
   })(
   /** @augments {React.Component<{k8sModels?: Map<string, K8sKind}>} */
   class Firehose extends React.Component {
-    componentWillMount (props=this.props) {
-      const { watchK8sList, watchK8sObject, resources, k8sModels, inFlight } = props;
+    // TODO: Convert this to `componentDidMount`
+    // eslint-disable-next-line camelcase
+    UNSAFE_componentWillMount () {
+      this.start();
+    }
+
+    componentWillUnmount () {
+      this.clear();
+    }
+
+    shouldComponentUpdate(nextProps) {
+      const {forceUpdate = false} = this.props;
+      if (nextProps.inFlight !== this.props.inFlight && nextProps.loaded) {
+        return forceUpdate;
```

Comment:
It looks to me like this will return false more often than before. Is that the intent?

I noticed the catalog page doesn't load with these changes. (@dtaylor113 is working on integration tests that would catch this.)

`shouldComponentUpdate` returning false has caused us some heartburn. @TheRealJon had to add `forceUpdate` for the overview. I wonder if we shouldn't profile this to see how much different it's making.